develop/reference_workarounds.rst

```diff
@@ -128,3 +128,201 @@ There is another trick though. We can pull the relevant information to the coord
+
+Suppose that for every value inserted into ``little_vals`` we want to insert one twice as big into ``big_vals``.
+
+.. code-block:: postgresql
+
+  CREATE TABLE little_vals (key int, val int);
+  CREATE TABLE big_vals    (key int, val int);
+  SELECT create_distributed_table('little_vals', 'key');
+  SELECT create_distributed_table('big_vals',    'key');
+
+  -- This trigger function takes the destination placement as an argument
+
+  SELECT run_command_on_workers($cmd$
+    CREATE OR REPLACE FUNCTION embiggen() RETURNS TRIGGER AS $$
+      BEGIN
+        IF (TG_OP = 'INSERT') THEN
+          EXECUTE format(
+            'INSERT INTO %s (key, val) SELECT ($1).key, ($1).val*2;',
+            TG_ARGV[0]
+          ) USING NEW;
+        END IF;
+        RETURN NULL;
+      END;
+    $$ LANGUAGE plpgsql;
+  $cmd$);
+
+  -- Next we relate the co-located tables by the trigger function
+  -- on each co-located placement
+
+  SELECT run_command_on_colocated_placements(
+    'little_vals',
+    'big_vals',
+    $cmd$
+      CREATE TRIGGER after_insert AFTER INSERT ON %s
+        FOR EACH ROW EXECUTE PROCEDURE embiggen(%s)
+    $cmd$
+  );
+
+Then to test it:
+
+.. code-block:: postgresql
+
+  INSERT INTO little_vals VALUES (1, 42), (2, 101);
+  TABLE big_vals;
+
+::
+
+  ┌─────┬─────┐
+  │ key │ val │
+  ├─────┼─────┤
+  │   1 │  84 │
+  │   2 │ 202 │
+  └─────┴─────┘
+
+**Trigger between reference tables.**
+
+Reference tables are simpler than distributed tables in that they have exactly one shard which is replicated across all workers. To relate reference tables with a trigger, we simply need to create the trigger for the shard on all workers.
+
+Suppose we want to record every change of ``insert_target`` in ``audit_table``, both of which are reference tables.
+
+.. code-block:: postgresql
+
+  -- create the reference tables
+
+  CREATE TABLE insert_target (
+    value text
+  );
+  CREATE TABLE audit_table(
+    stamp timestamp,
+    value text
+  );
+  SELECT create_reference_table('insert_target');
+  SELECT create_reference_table('audit_table');
+
+  -- now find the shard id for each table
+
+  SELECT logicalrelid, shardid
+  FROM pg_dist_shard
+  WHERE logicalrelid IN
+    ('insert_target'::regclass, 'audit_table'::regclass);
+
+Include the shard ids (written below as "xxxxxx" and "yyyyyy") in custom queries:
+
+.. code-block:: postgresql
+
+  SELECT run_command_on_workers($cmd$
+    CREATE OR REPLACE FUNCTION process_audit() RETURNS TRIGGER AS $$
+      BEGIN
+        INSERT INTO audit_table_xxxxxx(stamp,value) VALUES (now(),'value');
+        RETURN NEW;
+      END;
+    $$ LANGUAGE plpgsql;
+  $cmd$);
+
+  SELECT run_command_on_workers($cmd$
+    CREATE TRIGGER emp_audit
+    AFTER INSERT OR UPDATE OR DELETE ON insert_target_yyyyyy
```

Comment:
Do we need the YYYY here for insert target?

Reply:
Yeah, it's being run on the worker so it can't see plain old insert_target. (Tried it to double check and got an error)